docs/book/src/developer/providers/contracts/infra-cluster.md

```diff
@@ -0,0 +1,516 @@
+</aside>
+
+## Rules (contract version v1beta1)
+
+| Rule                                                                      | Mandatory | Note                                                                |
+|---------------------------------------------------------------------------|-----------|---------------------------------------------------------------------|
+| [All resources scope]                                                     | Yes       |                                                                     |
+| [All resources `TypeMeta` and `ObjectMeta`field]                          | Yes       |                                                                     |
+| [All resources `APIVersion` field value]                                  | Yes       |                                                                     |
+| [InfraCluster, InfraClusterList resource definition]                      | Yes       |                                                                     |
+| [InfraCluster control plane endpoint]                                     | No        | Mandatory if control plane endpoint is not provided by other means. |
+| [InfraCluster failure domains]                                            | No        |                                                                     |
+| [InfraCluster initialization completed]                                   | Yes       |                                                                     |
+| [InfraCluster conditions]                                                 | No        |                                                                     |
+| [InfraCluster terminal failures]                                          | No        |                                                                     |
+| [InfraClusterTemplate, InfraClusterTemplateList resource definition]      | No        | Mandatory for ClusterClasses support                                |
+| [Externally managed infrastructure]                                       | No        |                                                                     |
+| [Multi tenancy]                                                           | No        | Mandatory for clusterctl CLI support                                |
+| [Clusterctl support]                                                      | No        | Mandatory for clusterctl CLI support                                |
+
+Note:
+- `All resources` refers to all the provider's resources "core" Cluster API interacts with;
+  In the context of this page: `InfraCluster`, `InfraClusterTemplate` and corresponding list types
+
+### All resources scope
+
+All resources MUST be namespace-scoped.
+
+### All resources `TypeMeta` and `ObjectMeta` field
+
+All resources MUST have the standard Kubernetes “type metadata” and “object metadata”
+
+### All resources `APIVersion` field value
+
+All resources `APIVersion` is a combination of API group and version.
```

Comment:
this reads a bit strange 
```suggestion
All resources MUST have an `APIVersion` value that is a combination of API group and version.
```

(But it's not like Kubernetes allows you to freely chose the value. It will just force you to use apiGroup/version)

I would drop this one

Reply:
fixed, PTAL